File: 800/2157A/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int freq[] = new int[101];
            
            for(int i=0;i<n;i++)
                freq[sc.nextInt()]++;
            
            int cnt = 0;
            for(int i=0;i<=100;i++)
            {
                if(freq[i] == 0 || freq[i] == i)
                    continue;
                
                cnt += (freq[i] < i)? freq[i] : freq[i]-i;
            }
            
            System.out.println(cnt);
        }
    }
}

File: 800/2157A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n;
        cin >> n;

        vector<int> freq(101, 0);

        for (int i = 0; i < n; i++) 
        {
            int x;
            cin >> x;
            freq[x]++;
        }

        int cnt = 0;
        for (int i = 0; i <= 100; i++)
        {
            if (freq[i] == 0 || freq[i] == i) 
                continue;

            if (freq[i] < i)
                cnt += freq[i];
            else
                cnt += freq[i] - i;
        }

        cout << cnt << "\n";
    }

    return 0;
}
